ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -370,4 +401,30 @@ mod test {
 
         assert_eq!(q_final_another_way.as_u128(), EXPECTED_Q_FINAL);
     }
+
+    #[test]
+    fn differences_are_zero() {
+        const ZKP_1_and_2_LEFT: [u128; 14] = [0, 0, 13, 17, 11, 25, 7, 11, 25, 17, 9, 22, 23, 3];
+        const ZKP_1_and_2_RIGHT: [u128; 14] = [0, 30, 16, 13, 25, 3, 6, 1, 12, 29, 30, 7, 7, 3];
+        const OUT_LEFT: u128 = 27;
+        const OUT_RIGHT: u128 = 0;
+
+        let zkp_left = make_chunks::<Fp31, 7>(&ZKP_1_and_2_LEFT);
+        let zkp_right = make_chunks::<Fp31, 7>(&ZKP_1_and_2_RIGHT);
+        let challenges = vec![Fp31::truncate_from(22u128), Fp31::truncate_from(17u128)];
```

Comment:
You can add the third proof and third challenge here as well to better exercise the logic, since the first and last proof are somewhat special logic. Third challenge is 30.

Reply:
Right now, we use a vector which does not support items of different length. I was thinking of adding the support for having a different length first proof in a follow up PR.